kite-data/kite-data-core/src/main/java/org/kitesdk/data/spi/filesystem/FileSystemView.java

```diff
@@ -89,7 +89,7 @@ private FileSystemView(FileSystemView<E> view, Constraints c) {
   public DatasetWriter<E> newWriter() {
     AbstractDatasetWriter<E> writer;
     if (dataset.getDescriptor().isPartitioned()) {
-      writer = new PartitionedDatasetWriter<E>(this);
+      writer = PartitionedDatasetWriter.newWriter(this);
```

Comment:
Sorry, I wasn't clear. I mean that perhaps we should have a single utility method to return a writer for a view (partitioned or not) rather than having one for the partitioned case and one for the single-file case. Definitely not a blocker.